crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -20,6 +21,90 @@ where
         unimplemented!()
     }
 
+    /// Finds a given trasaction by its hash.
+    pub(crate) async fn transaction_by_hash(
+        &self,
+        hash: H256,
+    ) -> EthResult<Option<reth_rpc_types::Transaction>> {
+        let tx_by_hash = match self.client().transaction_by_hash(hash) {
+            Ok(Some(item)) => item,
+            Ok(None) => return Ok(None),
+            Err(_) => return Err(EthApiError::EmptyRawTransactionData),
+        };
+
+        let tx: reth_rpc_types::Transaction = Transaction::from_recovered_with_block_context(
+            tx_by_hash.into_ecrecovered().unwrap(),
+            // this is just stubbed out for now still need to fully implement
+            H256::default(),
+            u64::default(),
+            U256::from(usize::from(Index::default())),
+        );
+        Ok(Some(tx))
+    }
+
+    /// Find a transaction at a specific index in a block, after finding the block by its hash.
+    pub(crate) async fn transaction_by_block_hash_and_index(
+        &self,
+        hash: H256,
+        index: Index,
+    ) -> EthResult<Option<reth_rpc_types::Transaction>> {
+        let block_by_hash = match self.client().block_by_hash(hash) {
+            Ok(item) => item,
+            Err(_) => return Err(EthApiError::UnknownBlockNumber),
+        };
+
+        if let Some(block) = block_by_hash {
+            let txs = block.body;
+            let tx = txs[usize::from(index)].clone();
+            Ok(Some(Transaction::from_recovered_with_block_context(
+                tx.into_ecrecovered().unwrap(),
+                hash,
+                block.header.number,
+                U256::from(usize::from(index)),
+            )))
+        } else {
+            Ok(None)
+        }
```

Comment:
this can be improved by using

`TransactionsProvider::transactions_by_block` directly

Reply:
fixed